enterprise/signals.py

```diff
@@ -106,6 +107,59 @@ def default_content_filter(sender, instance, **kwargs):     # pylint: disable=un
         instance.save()
 
 
+@receiver(post_save, sender=EnterpriseCustomerUser)
+def assign_or_delete_enterprise_admin_role(sender, instance, **kwargs):     # pylint: disable=unused-argument
+    """
+    Assign or delete enterprise_admin role for EnterpriseCustomerUser when updated.
+
+    This only occurs if a PendingEnterpriseCustomerAdminUser record exists.
+    """
+    if instance.user:
+        enterprise_admin_role, __ = SystemWideEnterpriseRole.objects.get_or_create(name=ENTERPRISE_ADMIN_ROLE)
+        try:
+            pending_enterprise_admin_user = PendingEnterpriseCustomerAdminUser.objects.get(
+                user_email=instance.user.email,
+                enterprise_customer=instance.enterprise_customer,
+            )
+        except PendingEnterpriseCustomerAdminUser.DoesNotExist:
+            pending_enterprise_admin_user = None
+
+        if kwargs['created'] and pending_enterprise_admin_user:
+            # EnterpriseCustomerUser record was created and a pending admin user
+            # exists, so assign the enterprise_admin role.
+            pending_enterprise_admin_user.activate_admin_permissions(
+                user=instance.user,
+                enterprise_customer=instance.enterprise_customer,
+            )
+        elif not kwargs['created'] and not instance.linked:
+            # EnterpriseCustomerUser record was updated but is not linked, so delete the enterprise_admin role
+            try:
+                SystemWideEnterpriseUserRoleAssignment.objects.get(
+                    user=instance.user,
+                    role=enterprise_admin_role
+                ).delete()
+            except SystemWideEnterpriseUserRoleAssignment.DoesNotExist:
+                # Do nothing if no role assignment is present for the enterprise customer user.
+                pass
+
+
+@receiver(post_delete, sender=EnterpriseCustomerUser)
+def delete_enterprise_admin_role_assignment(sender, instance, **kwargs):     # pylint: disable=unused-argument
+    """
+    Delete the associated enterprise admin role assignment record when deleting an EnterpriseCustomerUser record.
+    """
+    if instance.user:
+        enterprise_admin_role, __ = SystemWideEnterpriseRole.objects.get_or_create(name=ENTERPRISE_ADMIN_ROLE)
+        try:
+            SystemWideEnterpriseUserRoleAssignment.objects.get(
+                user=instance.user,
+                role=enterprise_admin_role
+            ).delete()
+        except SystemWideEnterpriseUserRoleAssignment.DoesNotExist:
+            # Do nothing if no role assignment is present for the enterprise customer user.
+            pass
```

Comment:
It shouldn't happen often, but you can technically have multiple role assignments for the same role and user so I'd suggest we use `filter` instead of `get`
  
![image](https://user-images.githubusercontent.com/14864970/92620390-965b4e00-f290-11ea-961d-34f0f782d4aa.png)